### PR TITLE
optimizations: add TODOs, make some functionality async

### PR DESCRIPTION
**Pull request by noot**
## Changes

- add TODO comments
- make digest handling async
- make writing state trie after executing block async
- 1 block now takes ~13s

## Tests

```
go test ./... -short
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] All integration tests and required coverage checks are passing

## Issues

-

### CODE REVIEW

**Review by edwardmack — APPROVED** [review]

File: dot/sync/syncer.go [hunk 1]
```diff
@@ -469,14 +471,14 @@ func (s *Service) handleBlock(block *types.Block) error {
...
-		err = s.handleDigests(block.Header)
-		if err != nil {
-			return err
-		}
+		go func() {
+			err = s.handleDigests(block.Header)
+			if err != nil {
+				s.logger.Error("failed to handle block digest", "error", err)
+			}
+		}()
```
Comment (anthdm) [inline dot/sync/syncer.go:481]: Same here, can we just let the error do its thing?

File: dot/state/storage.go [hunk 2]
```diff
@@ -114,13 +114,17 @@ func (s *StorageState) StoreTrie(root common.Hash, ts *rtstorage.TrieState) erro
...
+	go func() {
+		err = s.StoreInDB(root)
+		if err != nil {
+			logger.Error("failed to store encoded trie in database", "state root", root, "error", err)
+			return
+		}
 
-	logger.Trace("stored trie in database", "root", root)
+		logger.Trace("stored trie in database", "root", root)
+	}()
```
Comment (anthdm) [inline dot/state/storage.go:127]: Are we not depending on the possible error that `StoreInDB` could produce?

Comment (noot) [reply dot/state/storage.go:127]: we aren't really depending on it, this error generally shouldn't happen unless there's something wrong with the database, like it's been closed. if we want to handle the error we'd have to add some sort of err channel to get it out

File: dot/sync/syncer.go (hunk 1 above)
Comment (noot) [reply dot/sync/syncer.go:481]: similar to below, if it errors it's not good but the node wouldn't crash or anything

**Review by RyRy79261 — APPROVED** [review]

**Pull request merged**